scripts/create_newcase

```diff
@@ -1,1003 +1,177 @@
+
+    parser.add_argument("--user-grid", action="store_true",
+                        help="If set, then the -grid argument is treated as a user specified grid."
+                        "As such, the additional argument --gridfile must also be specified"
+                        "and all of the grid settings in the --res argument name (which must be a grid longname"
+                        "have been defined in the grid file pointed to by --gridfile")
+
+    parser.add_argument("--gridfile",
+                        help="Full pathname of config grid file to use"
+                        "This should be a copy of cime_config/config_grids.xml"
+                        "with the new user grid changes added to it"
+			"This argument is required if --user-grid is True")
+
+    parser.add_argument("--srcroot",
+                        help="Alternative path for source root directory. By default this is set to"
+                        "cimeroot/../")
+
+    # hidden argument indicating called from create_test
+    parser.add_argument("--test", "-test", action="store_true",
+                        help="Used to indicate that create_newcase was called from create_test"
+                        "- do not use otherwise")
+
+    args = parser.parse_args()
+
+    CIME.utils.handle_standard_logging_options(args)
+
+    if args.srcroot is not None:
+        expect(os.path.isdir(args.srcroot),
+               "Input non-default directory srcroot %s does not exist " %args.srcroot)
+        args.srcroot = os.path.abspath(args.srcroot)
+
+    if args.gridfile is not None:
+        expect(os.path.isfile(args.gridfile),
+               "Grid specification file %s does not exist " %args.gridfile)
+
+    if args.pesfile is not None:
+        expect(os.path.isfile(args.pesfile),
+               "Pes specification file %s cannot be found " %args.pesfile)
+
+    if args.user_compset:
+        expect(args.pesfile is not None,
+               "--pesfile is required if --user-compset is set")
+
+    if args.user_grid:
+        expect(args.gridfile is not None,
+               "User grid specification file must be set if the user grid is requested")
+
+    return args.case, args.compset, args.res, args.mach, args.compiler,\
+        args.mpilib, args.project, args.pecount, \
+        args.mach_dir, args.user_mods_dir, args.user_compset, args.pesfile, \
+        args.user_grid, args.gridfile, args.srcroot, args.test
+
+
+###############################################################################
+def _main_func():
+###############################################################################
+
+    case, compset, grid, machine, compiler, \
+        mpilib, project, pecount,  \
+        machine_dir, usermods_dir, user_compset, pesfile, \
+        user_grid, gridfile, srcroot, test \
+        = parse_command_line(sys.argv)
+
+    cimeroot  = os.path.abspath(CIME.utils.get_cime_root())
+
+    if srcroot is None:
```

Comment:
You could save yourself some work here by just defaulting srcroot to os.path.dirname(cimeroot) in the argparse setup above.